src/IO/S3/copyS3File.cpp

```diff
@@ -149,15 +149,20 @@ namespace
                                            dest_bucket, dest_key, /* local_path_ */ {}, /* data_size */ 0,
                                            outcome.IsSuccess() ? nullptr : &outcome.GetError());
 
-            if (outcome.IsSuccess())
+            if (!outcome.IsSuccess())
             {
-                multipart_upload_id = outcome.GetResult().GetUploadId();
-                LOG_TRACE(log, "Multipart upload has created. Bucket: {}, Key: {}, Upload id: {}", dest_bucket, dest_key, multipart_upload_id);
+                ProfileEvents::increment(ProfileEvents::WriteBufferFromS3RequestsErrors, 1);
+                throw S3Exception(outcome.GetError().GetMessage(), outcome.GetError().GetErrorType());
             }
-            else
+            else if (outcome.GetResult().GetUploadId().empty())
             {
                 ProfileEvents::increment(ProfileEvents::WriteBufferFromS3RequestsErrors, 1);
-                throw S3Exception(outcome.GetError().GetMessage(), outcome.GetError().GetErrorType());
+                throw Exception(ErrorCodes::S3_ERROR, "Invalid CreateMultipartUpload result: missing UploadId.");
+            }
+            else
+            {
+                multipart_upload_id = outcome.GetResult().GetUploadId();
```

Comment:
While reading this diff I stumbled upon the fact that GetResult is called twice. Does it have side effects?

Reply:
No, it just returns a reference to a field.